torch/csrc/distributed/rpc/process_group_agent.h

```diff
@@ -64,6 +64,12 @@ class ProcessGroupAgent : public RpcAgent {
   void enqueueRecv(RecvWork work);
   // receiving messages
   void listenLoop();
+  // globally checks if there is no pending messages. This method reads send and
+  // recv counters, and then use an allgather to collect all counters from all
+  // peers, i.e., all workers will have the same view of the counters. Then, a
+  // worker will only return true if all send counters match all receive
+  // coutners.
```

Comment:
Typo in the summary. Also I think this can benefit from a bigger explanation of how the algorithm works.